samples/oneapi/dpc/mpi/sources/decision_forest_cls_hist_distr_mpi.cpp

```diff
@@ -17,106 +17,103 @@
-              << result_train.get_var_importance() << std::endl;
-
-    std::cout << "OOB error: " << result_train.get_oob_err() << std::endl;
-  }
-
-  const auto result_infer = dal::preview::infer(
-      comm, df_desc, result_train.get_model(), x_test_vec[rank_id]);
-
-  if (comm.get_rank() == 0) {
-    std::cout << "Prediction results:\n"
-              << result_infer.get_responses() << std::endl;
-    std::cout << "Probabilities results:\n"
-              << result_infer.get_probabilities() << std::endl;
-
-    std::cout << "Ground truth:\n" << y_test_vec[rank_id] << std::endl;
-  }
+void run(sycl::queue &queue) {
+    const auto train_data_file_name =
+        get_data_path("df_classification_train_data.csv");
+    const auto train_response_file_name =
+        get_data_path("df_classification_train_label.csv");
+    const auto test_data_file_name =
+        get_data_path("df_classification_test_data.csv");
+    const auto test_response_file_name =
+        get_data_path("df_classification_test_label.csv");
+
+    const auto x_train =
+        dal::read<dal::table>(queue, dal::csv::data_source{train_data_file_name});
+    const auto y_train = dal::read<dal::table>(
+        queue, dal::csv::data_source{train_response_file_name});
+    const auto x_test =
+        dal::read<dal::table>(queue, dal::csv::data_source{test_data_file_name});
+    const auto y_test = dal::read<dal::table>(
+        queue, dal::csv::data_source{test_response_file_name});
+
+    auto comm =
+        dal::preview::spmd::make_communicator<dal::preview::spmd::backend::mpi>(
+            queue);
+    auto rank_id = comm.get_rank();
+    auto rank_count = comm.get_rank_count();
+
+    auto x_train_vec = split_table_by_rows<float>(queue, x_train, rank_count);
+    auto y_train_vec = split_table_by_rows<float>(queue, y_train, rank_count);
+    auto x_test_vec = split_table_by_rows<float>(queue, x_test, rank_count);
+    auto y_test_vec = split_table_by_rows<float>(queue, y_test, rank_count);
+
+    const auto df_desc =
+        df::descriptor<float, df::method::hist, df::task::classification>{}
+            .set_class_count(5)
+            .set_tree_count(10)
+            .set_features_per_node(x_train.get_column_count())
+            .set_min_observations_in_leaf_node(8)
+            .set_min_observations_in_split_node(16)
+            .set_min_weight_fraction_in_leaf_node(0.0)
+            .set_min_impurity_decrease_in_split_node(0.0)
+            .set_error_metric_mode(df::error_metric_mode::out_of_bag_error)
+            .set_variable_importance_mode(df::variable_importance_mode::mdi)
+            .set_infer_mode(df::infer_mode::class_responses |
+                            df::infer_mode::class_probabilities)
+            .set_voting_mode(df::voting_mode::weighted);
+
+    const auto result_train = dal::preview::train(
+        comm, df_desc, x_train_vec[rank_id], y_train_vec[rank_id]);
+
+    if (comm.get_rank() == 0) {
```

Comment:
```suggestion
     if (rank_id == 0) {
```

Reply:
changed all conditions to if (comm.get_rank() == 0) { to get it in one style